File: src/hackerrank/java/BonAppetit.java
package hackerrank.java;

import java.util.ArrayList;
import java.util.List;

public class BonAppetit {

    /**
     * Default test case
     */
    public static void main(String[] args) {
        List<Integer> bill = new ArrayList<>() {{
            add(3);
            add(10);
            add(2);
            add(9);
        }};

        bonAppetit(bill, 1, 12);
    }

    static void bonAppetit(List<Integer> bill, int k, int b) {

        int sum = bill.stream().mapToInt(Integer::intValue).sum();

        int paid = (sum - bill.get(k)) / 2;
        if (paid == b) {
            System.out.println("Bon Appetit");
        } else {
            System.out.println(b - paid);
        }
    }
}

File: src/hackerrank/kotlin/BonAppetit.kt
package hackerrank.kotlin

fun main(args: Array<String>) {
    bonAppetit(intArrayOf(3, 10, 2, 9).toTypedArray(), 1, 12)
}

fun bonAppetit(bill: Array<Int>, k: Int, b: Int) = ((bill.sum() - bill[k]) / 2).let {
    println(if (it == b) "Bon Appetit" else b - it)
}